src/Microsoft.AspNet.Razor.Runtime/BufferedStringCollection.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Implements <see cref="IBufferedStringCollection"/> to buffer a collection of strings.
+    /// </summary>
+    public class BufferedStringCollection : IBufferedStringCollection
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="BufferedStringCollection"/>
+        /// </summary>
+        public BufferedStringCollection()
+        {
+            Buffer = new BufferEntryCollection();
+        }
+
+        /// <summary>
+        /// A collection of entries buffered by this instance of <see cref="BufferedStringCollection"/>.
+        /// </summary>
+        public BufferEntryCollection Buffer { get; private set; }
+
+        /// <inheritdoc />
+        public void Append(string value)
```

Comment:
We decided on `Append()` because that sounds better when `string`s are used. This method is mostly used in `TagHelper.Content / PreContent / PostContent`.